src/apps/filters/kRing.c

```diff
@@ -17,43 +17,57 @@
  * @brief stdin/stdout filter that converts from integer H3 indexes to
  * k-rings
  *
- *  usage: `kRing [k]`
+ *  usage: `kRing [k] [printDistances]`
  *
  *  The program reads H3 indexes from stdin until EOF and outputs
  *  the H3 indexes within k-ring `k` to stdout.
+ *
+ *  `printDistances` may be specified to also print the grid distance
+ *  from the origin index. Valid values are 0 to not print distances
+ *  (default) or 1 to print distances.
  */
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include "algos.h"
-#include "h3Index.h"
-#include "stackAlloc.h"
+#include "h3api.h"
 #include "utility.h"
 
-void doCell(H3Index h, int k) {
+void doCell(H3Index h, int k, int printDistances) {
     int maxSize = H3_EXPORT(maxKringSize)(k);
     H3Index* rings = calloc(maxSize, sizeof(H3Index));
-    H3_EXPORT(kRing)(h, k, rings);
+    int* distances = calloc(maxSize, sizeof(int));
+    H3_EXPORT(kRingDistances)(h, k, rings, distances);
 
     for (int i = 0; i < maxSize; i++) {
         if (rings[i] != 0) {
-            h3Println(rings[i]);
+            h3Print(rings[i]);
+            if (printDistances) {
+                printf(" %d\n", distances[i]);
+            } else {
+                printf("\n");
+            }
         }
     }
 
+    free(distances);
     free(rings);
 }
 
 int main(int argc, char* argv[]) {
     // check command line args
-    if (argc != 2) {
-        fprintf(stderr, "usage: %s [k]\n", argv[0]);
+    if (argc != 2 && argc != 3) {
+        fprintf(stderr, "usage: %s [k] [printDistances]\n", argv[0]);
         exit(1);
     }
 
     int k = 0;
-    if (argc > 1) {
-        if (!sscanf(argv[1], "%d", &k)) error("k must be an integer");
+    if (!sscanf(argv[1], "%d", &k)) error("k must be an integer");
+
+    int printDistances = 0;
+    if (argc > 2) {
+        if (!sscanf(argv[2], "%d", &printDistances))
+            error("printDistances must be an integer");
```

Comment:
This seems weird? It's a boolean flag, so ideally we'd want something like `--printDistances` or `-p`. This would be valid for values > 1 as well, which seems odd. Any reason not to use `strcmp` here and have more descriptive flags?

Reply:
I'll address this in a second PR since adding that functionality is a little complicated. For the moment this is using the same pattern as selecting KML output with positional arguments.